openedx/frontend-app-course-authoring
# feat: [FC-0044] Course unit page - Display sidebar component

**Settings**

```yaml
EDX_PLATFORM_REPOSITORY: https://github.com/raccoongang/edx-platform.git
EDX_PLATFORM_VERSION: ruzniaievdm/feat/unit-children

TUTOR_GROVE_WAFFLE_FLAGS:
  - name: contentstore.new_studio_mfe.use_new_unit_page
    everyone: true

TUTOR_GROVE_MFE_LMS_COMMON_SETTINGS:
  MFE_CONFIG:
    ENABLE_UNIT_PAGE: true
```

### Description

This pull request added a sidebar for the Course unit page and an alert notifying about unpublished changes.

The primary features were implemented:
- A new Sidebar component.
- Alert notifying about unpublished changes.

**Issue:** https://github.com/openedx/platform-roadmap/issues/321

### Developer notes
- An updated visual representation of the Copy Unit button will be added after adding the copy-paste logic for units and xblocks.
- The initial 1 commit in this pull request is temporary. Once the https://github.com/openedx/frontend-app-course-authoring/pull/857 is merged, the third commit of this pull request will become the main commit.

### Design
[Figma design](https://www.figma.com/file/YeKFwSpyLaJFDs3f3p3TSa/Studio-1%3A1-mock-ups?type=design&node-id=599-23595&mode=design&t=icsTrD5n70NSEnND-0)

<img width="452" alt="image" src="https://github.com/openedx/frontend-app-course-authoring/assets/93188219/ebc3ad61-89c6-4c1d-9239-dc3e92c73801">

### Testing instructions
1. Run master devstack.
2. Start platform make dev.up.lms+cms+frontend-app-course-authoring and make checkout on this branch.
3. Enable the new Unit page by adding a waffle flag `contentstore.new_studio_mfe.use_new_unit_page` in the CMS admin panel.
4. Make sure that the MFE setting `ENABLE_UNIT_PAGE=true` is enabled.
5. Go to the Course Unit page from the Course Outline page.
6. Make sure the course you are viewing is not outdated.
7. Publish all sections on the Course Outline page.

https://github.com/openedx/frontend-app-course-authoring/assets/93188219/b0606920-19da-48cb-92d7-f8311dd37514
